### PR TITLE
NMS-8751: Poller Node Down without outage

### PR DESCRIPTION
**Pull request by rssntn67**
NMS-8751: Poller Node Down without outage

* JIRA: http://issues.opennms.org/browse/NMS-8751

### CODE REVIEW

File: opennms-services/src/main/java/org/opennms/netmgt/poller/PollerEventProcessor.java [hunk 1]
```diff
@@ -726,6 +726,8 @@ private void serviceReschedule(Long nodeId, String nodeLabel, Event sourceEvent,
         for (final Service databaseService : databaseServices) {
             if (polledServices.contains(databaseService)) {
                 LOG.debug("{} is being skipped. Already scheduled.", databaseService);
+                //FIXME Discuss with Jesse
```
Comment (j-white) [inline opennms-services/src/main/java/org/opennms/netmgt/poller/PollerEventProcessor.java]: Let's discuss. It looks like I may have forgot the continue statement when I made the changes in NMS-7761.

File: opennms-services/src/main/java/org/opennms/netmgt/poller/pollables/PollableElement.java [hunk 2]
```diff
@@ -526,6 +526,8 @@ protected void doInheritParentalCause() {
             // I have no cause but my parent is down.. mark me as down as well
             // I already have a cause that's larger than myself then inherit as well
             setCause(parentalCause);
+            if (getStatus().isUp())
+            	createOutage(parentalCause);
```
Comment (j-white) [inline opennms-services/src/main/java/org/opennms/netmgt/poller/pollables/PollableElement.java]: Can you elaborate on why this was necessary?

File: opennms-services/src/main/java/org/opennms/netmgt/poller/QueryManagerDaoImpl.java [hunk 3]
```diff
@@ -318,4 +318,33 @@ public void updateServiceStatus(int nodeId, String ipAddr, String serviceName, S
                     serviceName, nodeId,  ipAddr, status, e);
         }
     }
+    
+    @Override
+    public void updateServiceLastGood(int nodeId, String ipAddr, String serviceName, Date updateTime) {
+        try {
+            OnmsMonitoredService service = m_monitoredServiceDao.get(nodeId, InetAddress.getByName(ipAddr), serviceName);
+            service.setLastGood(updateTime);;
```
Comment (j-white) [inline opennms-services/src/main/java/org/opennms/netmgt/poller/QueryManagerDaoImpl.java]: Is updating the lastGood and lastFail columns only informational, or do they have some formal purpose?

I noticed the columns were already in the database, but appeared to be unused prior to this.

**Comment by j-white** [comment]
We should also add test cases for the scenarios described in the JIRA issue.

File: opennms-services/src/main/java/org/opennms/netmgt/poller/PollerEventProcessor.java (hunk 1 above)
Comment (j-white) [inline opennms-services/src/main/java/org/opennms/netmgt/poller/PollerEventProcessor.java]: While investigating another pollerd related bug I came across this same problem. See https://github.com/OpenNMS/opennms/pull/1311.

**Comment by rssntn67** [comment]
I think you do this in the other pull request. Let me close without merging

**Pull request closed without merging**